File: 259. 3Sum Smaller.java
class Solution {
    public int threeSumSmaller(int[] nums, int target) {
        if (nums==null || nums.length<3) {
            return 0;
        }
        Arrays.sort(nums);
        int ans = 0;
        for (int i=0; i<nums.length; i++) {
            int newt = target - nums[i];
            int lo = i+1, hi = nums.length-1;
            while (lo < hi) {
                if (nums[lo] + nums[hi] < newt) {
                    ans += hi-lo;
                    lo++;
                } else {
                    hi--;
                }
            }
        }
        return ans;
    }
}

File: 259. 3Sum Smaller.py
class Solution:
    def threeSumSmaller(self, nums: List[int], target: int) -> int:
        nums.sort()
        count = 0
        for i in range(len(nums)):
            newtar = target - nums[i]
            lo, hi = i+1, len(nums)-1
            while lo < hi:
                if nums[lo] + nums[hi] >= newtar:
                    hi-=1
                else:
                    count+=hi-lo
                    lo+=1
        return count